xbmc/peripherals/bus/virtual/PeripheralBusAddon.h

```diff
@@ -34,12 +34,18 @@
 
 namespace PERIPHERALS
 {
-  class CPeripheralBusAddon : public CPeripheralBus
+  class CPeripheralBusAddon : public CPeripheralBus, public ADDON::IBinaryAddonManagerCallback
```

Comment:
I don't really like this. IMO nothing outside of CAddonManager should know that CBinaryAddonManager exists and there shouldn't be separate events or notifications depending on which type of addon is used. It completely breaks encapsulation and makes things very messy.

I suggest fixing the design so the events go through the existing CAddonManager events mechanism that we already have instead of trying to replace it.